src/charm.py

```diff
@@ -109,6 +136,61 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         container.replan()
         self._update_app_and_unit_status(ActiveStatus())
 
+    def _database_uri(self) -> typing.Dict[str, str]:
+        """Compute DatabaseURI and return it.
+
+        Returns:
+            DatabaseURI containing details about the data provider integration
+        """
+        db_uris: typing.Dict[str, str] = {}
+
+        # the database_requirer could not be defined
+        # if _database_uri() is called before its initialization
+        if not hasattr(self, "databases") or not self.databases:
+            return db_uris
+
+        for database, db_requires in self.databases.items():
+            relation_data = list(db_requires.fetch_relation_data().values())
+
+            if not relation_data:
+                continue
+
+            # There can be only one database integrated at a time
+            # see: metadata.yaml
+            data = relation_data[0]
+
+            # Check that the relation data is well formed according to the following json_schema:
+            # https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/mysql_client/v0/schemas/provider.json
+            if not all(data.get(key) for key in ("endpoints", "username", "password")):
```

Comment:
Should this be an exception that puts the charm in blocked state?

Reply:
Done